src/components/shared/thread/SelectedActivityGraph.js

```diff
@@ -71,138 +66,44 @@ type StateProps = {|
-    }
-    if (callNodeTable.category[parentCallNode] !== clickedCategory) {
-      break;
-    }
-    callNodesOnSameCategoryPath.push(parentCallNode);
-    callNode = parentCallNode;
-  }
-
-  // Now find the callNode in callNodesOnSameCategoryPath with the lowest depth
-  // such that selecting it will not highlight any samples whose unfiltered
-  // category is different from clickedCategory. If no such callNode exists,
-  // return clickedCallNode.
-
-  const handledCallNodes = new Uint8Array(callNodeTable.length);
-  function limitSameCategoryPathToCommonAncestor(callNode) {
-    const walkUpToDepth =
-      clickedDepth - (callNodesOnSameCategoryPath.length - 1);
-    let depth = callNodeTable.depth[callNode];
-    while (depth >= walkUpToDepth) {
-      if (handledCallNodes[callNode]) {
-        return;
-      }
-      handledCallNodes[callNode] = 1;
-      if (depth <= clickedDepth) {
-        if (callNode === callNodesOnSameCategoryPath[clickedDepth - depth]) {
-          callNodesOnSameCategoryPath.length = clickedDepth - depth;
-          return;
-        }
-      }
-      callNode = callNodeTable.prefix[callNode];
-      depth--;
-    }
-  }
-
-  for (let sample = 0; sample < sampleCallNodes.length; sample++) {
-    if (
-      sampleCategories[sample] !== clickedCategory &&
-      sampleCallNodes[sample] !== null
-    ) {
-      limitSameCategoryPathToCommonAncestor(sampleCallNodes[sample]);
-    }
-  }
-
-  if (callNodesOnSameCategoryPath.length > 0) {
-    return callNodesOnSameCategoryPath[callNodesOnSameCategoryPath.length - 1];
-  }
-  return clickedCallNode;
-}
-
-function getSampleCategories(
-  samples: SamplesTable,
-  stackTable: StackTable
-): Array<IndexIntoSamplesTable | null> {
-  return samples.stack.map(s => (s !== null ? stackTable.category[s] : null));
-}
-
 class SelectedThreadActivityGraphCanvas extends PureComponent<Props> {
-  _onSampleClick = (sampleIndex: IndexIntoSamplesTable) => {
+  /**
+   *
+   */
+  _onActivitySampleClick = (sampleIndex: IndexIntoSamplesTable) => {
     const {
-      fullThread,
-      filteredThread,
-      callNodeInfo,
       selectedThreadIndex,
-      changeSelectedCallNode,
+      selectBestAncestorCallNodeAndExpandCallTree,
       focusCallTree,
     } = this.props;
-    const unfilteredStack = fullThread.samples.stack[sampleIndex];
-    if (unfilteredStack === null) {
-      return;
-    }
-
-    const clickedCategory = fullThread.stackTable.category[unfilteredStack];
-    const { callNodeTable, stackIndexToCallNodeIndex } = callNodeInfo;
-    const sampleCallNodes = getSampleCallNodes(
-      filteredThread.samples,
-      stackIndexToCallNodeIndex
+    const isBestCallNodeFound = selectBestAncestorCallNodeAndExpandCallTree(
+      selectedThreadIndex,
+      sampleIndex
     );
-    const clickedCallNode = sampleCallNodes[sampleIndex];
-    if (clickedCallNode === null) {
-      return;
+    if (isBestCallNodeFound) {
+      focusCallTree();
```

Comment:
I'm actually not sure if skipping the call to `focusCallTree()` was intentional. It might make sense to just always focus it if the selected thread activity graph is clicked.

Reply:
Works for me.